### ISSUE TITLE
cache: serialize data emit

### ISSUE THREAD

**Issue by salewski** [salewski/ads-github-tools#85]
```console
$ ads-github-cache --version
ads-github-cache 0.3.4  (built: 2022-10-16 20:39:02)

Copyright (C) 2020, 2021, 2022 Alan D. Salewski <ads@salewski.email>
License GPLv2+: GNU GPL version 2 or later <http://gnu.org/licenses/gpl.html>.
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Written by Alan D. Salewski.
```

In issue #80 and issue #81, the `ads-github-cache` tool was given the ability to concurrently fetch data from the remote GitHub v3 API, which sped up the `update` operation considerably. However, we inadvertently broke operations that emitted the cached data, when operating in the "online" cache mode (which basically follows a pull-data-then-print approach). This can result in some of the cached data being emitted by several processes simultaneously, with the output intermixed (and therefore broken). Such behavior can sometimes be observed by simply piping the JSON data from an endpoint path into the `jq` tool:

```console
    $ ads-github-cache --get-cached '/user/repos' | jq '.'
    parse error: Invalid numeric literal at line 1, column 94214
```

The problem noted in the `jq` error message is that the JSON structure has been corrupted; the particular error could be different, depending on the how the output happens to be intermixed on any given run of the program.

If we artificially limit the number of background processes to one, then we effectively serialize the output and works around the problem.

```console
    $ ads-github-cache -j 1 --get-cached '/user/repos' | jq '.'
    <works>
```

**Comment by salewski** [comment]
I labeled this bug with `severity: 4 (important)` because workarounds in downstream consuming tools (such as `ads-github-fetch-all-upstreams`) are painful -- it has a major effect on the usability of the software for most users.

**Comment by salewski** [comment]
The fix for this issue has been merged to the `master` branch, and will be part of the `ads-github-tools-0.3.5` release.

**Comment by salewski** [comment]
This issue has been fixed for `ads-github-tools-0.3.5` (released `2022-10-26`).

[The blurb about it](https://github.com/salewski/ads-github-tools/blob/master/NEWS#issue-85-cache-serialize-data-emit) from our ['NEWS'](https://github.com/salewski/ads-github-tools/blob/master/NEWS) file:

```
*** issue #85: "cache: serialize data emit"

    https://github.com/salewski/ads-github-tools/issues/85

    The concurrency features added to the 'ads-github-cache' tool in the
    ads-github-tools-0.3.4 release (see issues #80 and #81) inadvertently
    broke some operations that emitted cached data. Data was not only
    concurrently fetched for updating the cache, but was sometimes emitted
    concurrently, as well, causing the output to be garbled (mainly in the
    form of JSON that was no longer well-formed due to being intermixed from
    multiple emitting background processes). This has been fixed. The data is
    still fetched concurrently (from the upstream GitHub v3 API endpoints),
    but emitting the data from cached objects now happens sequentially as one
    would expect. This is mainly relevant for paths that represent "paged
    collections", where multiple pages of data are available.
```
